src/Avalonia.Base/Input/Platform/KeyGestureFormatInfo.cs

```diff
@@ -0,0 +1,128 @@
+        
+        /// <summary>
+        /// Gets the most appropriate KeyGestureFormatInfo for the IFormatProvider requested.  This will be, in order:
+        /// 1. The provided IFormatProvider as a KeyGestureFormatInfo
+        /// 2. The currently registered platform specific KeyGestureFormatInfo, if present.
+        /// 3. The Invariant otherwise.
+        /// </summary>
+        /// <param name="formatProvider">The IFormatProvider to get a KeyGestureFormatInfo for.</param>
+        /// <returns></returns>
+        public static KeyGestureFormatInfo GetInstance(IFormatProvider? formatProvider)
+            => formatProvider?.GetFormat(typeof(KeyGestureFormatInfo)) as KeyGestureFormatInfo
+            ?? AvaloniaLocator.Current.GetService<KeyGestureFormatInfo>()
+            ?? Invariant;
+
+        /// <summary>
+        /// A dictionary of the common platform Key overrides. These are used as a fallback
+        /// if platformKeyOverrides doesn't contain the Key in question.
+        /// </summary>
+        
+        private static readonly Dictionary<Key, string> s_commonKeyOverrides = new()
+        {
+            { Key.Add , "+" },
+            { Key.D0 , "0" },
+            { Key.D1 , "1" },
+            { Key.D2 , "2" },
+            { Key.D3 , "3" },
+            { Key.D4 , "4" },
+            { Key.D5 , "5" },
+            { Key.D6 , "6" },
+            { Key.D7 , "7" },
+            { Key.D8 , "8" },
+            { Key.D9 , "9" },
+            { Key.Decimal , "." },
+            { Key.Divide , "/" },
+            { Key.Multiply , "*" },
+            { Key.OemBackslash , "\\" },
+            { Key.OemCloseBrackets , "]" },
+            { Key.OemComma , "," },
+            { Key.OemMinus , "-" },
+            { Key.OemOpenBrackets , "[" },
+            { Key.OemPeriod , "." },
+            { Key.OemPipe , "|" },
+            { Key.OemPlus , "+" },
+            { Key.OemQuestion , "/" },
+            { Key.OemQuotes , "\"" },
+            { Key.OemSemicolon , ";" },
+            { Key.OemTilde , "`" },
+            { Key.Separator , "/" },
+            { Key.Subtract , "-" }
+        };
+
+        /// <summary>
+        /// Checks the platformKeyOverrides and s_commonKeyOverrides Dictionaries, in order, for the appropriate
+        /// string to represent the given Key on this platform.
+        /// NOTE: If platformKeyOverrides is null, this is assumed to be the Invariant and the Dictionaries are not checked.
+        /// The plain Enum string is returned instead.
+        /// </summary>
+        /// <param name="key">The Key to format.</param>
+        /// <returns>The appropriate platform specific or common override if present, key.ToString() if not or this is the Invariant.</returns>
+        public string FormatKey(Key key)
+        {
+            /*
+             * The absence of an Overrides dictionary indicates this is the invariant, and
```

Comment:
This isn't always true, the constructor is public, so it's possible to pass a null dictionary that isn't the `Invariant` culture, and won't have fallback to the common keys.

Reply:
True, what I was intending to convey is that a null dictionary should be used as a flag to not override keys, as in the Invariant.  Is this an issue with my documentation wording, or is there a better way to implement that?  Like a bool parameter for "Use common overrides / use Key Enum strings" or some such?

Reply:
You documented the difference between null and an empty dictionary on the class itself so I'm fine with this behavior!